src/jacobian.jl

```diff
@@ -9,7 +9,7 @@ isinplace(JᵀJ::KrylovJᵀJ) = isinplace(JᵀJ.Jᵀ)
 
 # Select if we are going to use sparse differentiation or not
 sparsity_detection_alg(_, _) = NoSparsityDetection()
-function sparsity_detection_alg(f, ad::AbstractSparseADType)
+function sparsity_detection_alg(f::NonlinearFunction, ad::AbstractSparseADType)
```

Comment:
Does this need to specialize?

Reply:
safety measure, we are using `f.sparsity` and stuff in there.